src/team_lib/mhba/xccl_mhba_team.c

```diff
@@ -449,7 +453,7 @@ xccl_status_t xccl_mhba_team_destroy(xccl_tl_team_t *team)
         ibv_dereg_mr(mhba_team->dummy_bf_mr);
         free(mhba_team->work_completion);
         free(mhba_team->net.rank_map);
-        if (mhba_team->transpose_buf_mr) {
+        if (mhba_team->transpose) {
```

Comment:
this is not correct. Non -asr rank may have env variable set to 1 (mhba_team->transpose = 1) but they don't allocate transpose buf. Plz revert this change: previous was correct.